core/src/main/java/io/cucumber/core/plugin/JUnitFormatter.java

```diff
@@ -264,7 +264,7 @@ void handleEmptyTestCase(Document doc, Element tc, Result result) {
         private String calculateTotalDurationString(Result result) {
             DecimalFormat numberFormat = (DecimalFormat) NumberFormat.getNumberInstance(Locale.US);
             numberFormat.applyPattern("0.######");
-            return numberFormat.format(((double) result.getTimeDuration().toNanos()));
+            return numberFormat.format(((double) result.getDuration().toNanos() / 1000));
```

Comment:
I can't work out why this division is here.

Reply:
Owh wait. I'm looking at a single commit.

Reply:
So the old code was

```
return numberFormat.format(((double) result.getDuration()) / 1000000000);
```

Now it does:

```
return numberFormat.format(((double) result.getDuration().toNanos() / 1000));
```

What are you intending to do here?

Reply:
You asked to revert the test case back to the old millis values and keep the division correct? 

I reverted the test case and divided the nanos by `1000` to keep the values returned in millis like they were. If I just use `.toMillis()` it will return just round it to `0.0` instead of the correct `0.003`.

Reply:
I think the divisor should be 1000000000, and if you use millis in the test the expected result should stay the same.

In essence the reported time is in seconds.

Reply:
Ah ok. So I updated the TC to use `stepDuration = Duration.ofMillis(1L);` and the above code to do `((double) result.getDuration().toNanos() / 1000000000)` instead.

It keeps the expected result the same.